packages/pipeline/migrations/1550163069315-TokenOrderBookSnapshotsAddMakerAddress.ts

```diff
@@ -0,0 +1,36 @@
+import { MigrationInterface, QueryRunner, TableColumn } from 'typeorm';
+
+const TOKEN_ORDERBOOK_SNAPSHOT_TABLE = 'raw.token_orderbook_snapshots';
+const NEW_COLUMN_NAME = 'maker_address';
+
+export class TokenOrderBookSnapshotsAddMakerAddress1550163069315 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        const snapshotTable = await queryRunner.getTable(TOKEN_ORDERBOOK_SNAPSHOT_TABLE);
+        if (snapshotTable) {
+            const newColumn = new TableColumn({
+                name: NEW_COLUMN_NAME,
+                type: 'varchar',
+                isNullable: true,
+            });
+            await queryRunner.addColumn(TOKEN_ORDERBOOK_SNAPSHOT_TABLE, newColumn);
+            // backfill null values
+            await queryRunner.query(`
+                UPDATE ${TOKEN_ORDERBOOK_SNAPSHOT_TABLE}
+                SET ${NEW_COLUMN_NAME}='unknown'
+                WHERE ${NEW_COLUMN_NAME} is NULL;
+            `);
+            await queryRunner.query(`
+                ALTER TABLE ${TOKEN_ORDERBOOK_SNAPSHOT_TABLE}
+                    DROP CONSTRAINT "token_orderbook_snapshots_pkey1",
+                    ADD PRIMARY KEY (observed_timestamp, source, order_type, price, base_asset_symbol, quote_asset_symbol, maker_address);
```

Comment:
Did you know that there's an `updatePrimaryKeys()` method provided in typeorm?  Not sure it's functionally any different than what you're doing here, but thought you might like to know.  See how I used it in https://github.com/0xProject/0x-monorepo/blob/76d8a991c0fe5b9fc1371b85b851179d8d8daa01/packages/pipeline/migrations/1549499426238-AddTxHashToERC20ApprovalEventPrimaryKey.ts#L20

Reply:
Yea this didn't work for some reason.